File: src/main/java/com/tapadoo/slacknotifier/SlackProjectSettingsFactory.java
package com.tapadoo.slacknotifier;

import jetbrains.buildServer.serverSide.settings.ProjectSettings;
import jetbrains.buildServer.serverSide.settings.ProjectSettingsFactory;
import jetbrains.buildServer.serverSide.settings.ProjectSettingsManager;

/**
 * Created by jasonconnery on 02/03/2014.
 */
public class SlackProjectSettingsFactory implements ProjectSettingsFactory {

    public static final String SETTINGS_KEY = "slackSettings";

    public SlackProjectSettingsFactory(ProjectSettingsManager projectSettingsManager)
    {
        projectSettingsManager.registerSettingsFactory(SETTINGS_KEY, this);
    }

    public ProjectSettings createProjectSettings(String projectId) {
        return new SlackProjectSettings(projectId);
    }
}

File: src/main/kotlin/com/tapadoo/slacknotifier/SlackProjectSettingsFactory.kt
package com.tapadoo.slacknotifier

import jetbrains.buildServer.serverSide.settings.ProjectSettings
import jetbrains.buildServer.serverSide.settings.ProjectSettingsFactory
import jetbrains.buildServer.serverSide.settings.ProjectSettingsManager

/**
 * @author Jason Connery
 * @since  02/03/2014
 */
class SlackProjectSettingsFactory(projectSettingsManager: ProjectSettingsManager) : ProjectSettingsFactory {

  init {
    projectSettingsManager.registerSettingsFactory(SETTINGS_KEY, this)
  }

  override fun createProjectSettings(projectId: String): ProjectSettings = SlackProjectSettings(projectId)

  companion object {
    const val SETTINGS_KEY = "slackSettings"
  }
}
